server/datastore/mysql/apple_mdm.go

```diff
@@ -2875,29 +2875,106 @@ func subqueryOSSettingsStatusMac() (string, []any, error) {
 }
 
 func (ds *Datastore) GetMDMAppleProfilesSummary(ctx context.Context, teamID *uint) (*fleet.MDMProfilesSummary, error) {
-	subquery, args, err := subqueryOSSettingsStatusMac()
```

Comment:
I see this function is also used in `filterHostsByOSSettingsStatus`, do you know if we will have the same problems there?